tests/func/test_move.py

```diff
@@ -211,3 +213,34 @@ def test_should_move_to_dir_on_non_default_stage_file(tmp_dir, dvc):
     dvc.move("file", "directory")
 
     assert os.path.exists(os.path.join("directory", "file"))
+
+
+def test_move_gitignored(tmp_dir, scm, dvc):
+    from dvc.dvcfile import FileIsGitIgnored
+
+    tmp_dir.dvc_gen({"foo": "foo"})
+
+    os.mkdir("dir")
+    (tmp_dir / "dir").gen(".gitignore", "*")
+
+    with pytest.raises(FileIsGitIgnored):
+        dvc.move("foo", "dir")
+
+    assert (tmp_dir / "foo").read_text() == "foo"
+    assert (tmp_dir / "foo.dvc").exists()
+    assert not (tmp_dir / "dir" / "foo").exists()
+    assert not (tmp_dir / "dir" / "foo.dvc").exists()
+
+
+def test_move_output_overlap(tmp_dir, dvc):
+    from dvc.exceptions import OverlappingOutputPathsError
+
+    tmp_dir.dvc_gen({"foo": "foo", "dir": {"bar": "bar"}})
```

Comment:
Does this generate `dir.dvc` or `bar.dvc`? Should this test that `dir.dvc` does not change (and maybe that `dir/bar` still exists)?

Also, should both of these tests check that the `.gitignore` files haven't changed?

Reply:
This generates `dir.dvc`.

Regarding not making gitignore changes on failed stage creation, we check for that in other func tests, we shouldn't need to duplicate that here.